mlflow/tracking/client.py

```diff
@@ -186,13 +188,16 @@ def set_tag(self, run_id, key, value):
         tag = RunTag(key, str(value))
         self.store.set_tag(run_id, tag)
 
-    def delete_tag(self, run_id, key):
+    def delete_tag(self, id=None, key=None):
```

Comment:
@ankitmathur-db As we discussed offline, let's make this non-keyword args and remove the `is None` validation in line 197.